pkgs/applications/science/biology/neuron/default.nix

```
@@ -1,77 +1,107 @@
-, pkg-config
-, automake
-, autoconf
-, libtool
-, ncurses
 , readline
-, which
-, python ? null
-, useMpi ? false
+, xorg
 , mpi
-, iv
+, cmake
+, bison
+, flex
+, git
+, perl
+, gsl
+, xcbuild
+, python3
+, usempi ? false
+, useiv ? true
+, core ? false
+, rx3d ? false
 }:
-
 stdenv.mkDerivation rec {
-  pname = "neuron${lib.optionalString useMpi "-mpi"}";
-  version = "7.5";
-
-  nativeBuildInputs = [ which pkg-config automake autoconf libtool ];
-  buildInputs = [ ncurses readline python iv ]
-    ++ lib.optional useMpi mpi;
+  pname = "neuron";
+  version = "8.2.1";
+  enableParallelBuilding = true;
+  # format is for pythonModule conversion
+  format = "other";
+  propagatedNativeBuildInputs = [
+    readline
+  ] ++ (with lib;
+    optional useiv [
+      xorg.libX11
+      xorg.libX11.dev
+      xorg.libXcomposite
+      xorg.libXcomposite.dev
+      xorg.libXext
+      xorg.libXext.dev
+    ]
+    ++ optional usempi [
+      mpi
+    ]
+  );
+  nativeBuildInputs = ([
+    cmake
+    bison
+    flex
+    git
+    python3
+    python3.pkgs.wheel
+    python3.pkgs.setuptools
+    python3.pkgs.scikit-build
+    python3.pkgs.matplotlib
+  ]) ++ (with lib;
+    optional useiv ([])
+    ++ optional core ([
+      perl
+      gsl
+    ])
+    ++ optional usempi ([
+      python3.pkgs.mpi4py
+    ])
+    ++ optional rx3d ([
+      python3.pkgs.cython
+      python3.pkgs.numpy
+    ])
+    ++ optional stdenv.isDarwin ([
+      xcbuild
+    ]));
+  patches = [
+    (builtins.toFile "${pname}-${version}-DARWIN-PATCH" ''
+--- a/src/nrnpython/setup.py.in
++++ b/src/nrnpython/setup.py.in
+@@ -124,7 +124,7 @@ libdirs = [destdir + get_escaped_path("@NRN_LIBDIR@"),
+ rpath_prefix_flag='-Wl,-R'
+ extra_link_args = [@NRN_LINK_FLAGS_COMMA_SEPARATED_STRINGS@]
+ @MAC_DARWIN_FALSE@extra_link_args += [rpath_prefix_flag+lib_path for lib_path in libdirs]
+-@MAC_DARWIN_TRUE@extra_link_args.append("-Wl,-rpath,@loader_path/../../")
++@MAC_DARWIN_TRUE@extra_link_args.append("-Wl,-rpath,@loader_path/../../../")
+ @MAC_DARWIN_TRUE@extra_link_args.append("-Wl,-rpath,%s" % ivlibdir)
 
+ # as neuron module will be built during make, add build/lib
+'')
+  ];
+  postPatch = ''
+          patchShebangs --build .
+          sed -e 's#DESTDIR =#DESTDIR = '"$out"'#' -i external/coreneuron/extra/nrnivmodl_core_makefile.in
+        '';
+  cmakeFlags = [
+    ("-DNRN_ENABLE_INTERVIEWS=" + (if useiv then "ON" else "OFF"))
+    ("-DNRN_ENABLE_MPI=" + (if usempi then "ON" else "OFF"))
+    ("-DNRN_ENABLE_CORENEURON=" + (if core then "ON" else "OFF"))
+    ("-DNRN_ENABLE_RX3D=" + (if rx3d then "ON" else "OFF"))
+  ];
+  postInstall = ''
+          mkdir -p $out/${python3.sitePackages}
+          mv $out/lib/python/* $out/${python3.sitePackages}/
+          rm -rf $out/lib/python
+          rm -rf build
+          for entry in $out/lib/*.so; do
+            patchelf --set-rpath $(patchelf --print-rpath $entry | tr ':' '\n' | sed '/^\/build/d' | tr '\n' ':') $entry
+          done
```

```suggestion
    mkdir -p $out/${python3.sitePackages}
    mv $out/lib/python/* $out/${python3.sitePackages}/
    rm -rf $out/lib/python build
    for entry in $out/lib/*.so; do
      patchelf --set-rpath $(patchelf --print-rpath $entry | tr ':' '\n' | sed '/^\/build/d' | tr '\n' ':') $entry
    done
```

The patchelf is removing the build references, right? Maybe worth a short comment